### PR TITLE
fix: bundles are stored in job_history in their original format

### PR DESCRIPTION
**Pull request by crowecawcaw**
### What was the problem/requirement? (What/Why)
The bundle GUI submitter does not always save asset references or parameter values in the original file format when the bundle is saved to the `job_history` directory.

Job bundles have data files for the template, asset references and parameter values, and these files can be either JSON or YAML. If input job bundle has a JSON template, the saved bundle's template will also by JSON, but the asset references and parameter values will still be saved as YAML.

### What was the solution? (How)
Update the bundle saving logic so to that asset references and parameter values are saved in the same format as the job template.

### What is the impact of this change?
This unblocks some other work I'm doing on the Keyshot submitter where Keyshot's Python interpreter can only read and write JSON files. If saved bundle data is all in JSON, the submitter will be able to load it to find previously selected settings.

### How was this change tested?
I created a bundle with a JSON template, opened it with the GUI submitter, changed some parameters and job attachments, then exported the bundle. I verified that the new bundle in `job_history` had all JSON files. I also opened the GUI submitter for the new template and verified it loaded the JSON references and parameters correctly.

### Was this change documented?
No - bug fix.

### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/deadline/client/job_bundle/loader.py [hunk 1]
```diff
@@ -116,3 +118,20 @@ def read_yaml_or_json_object(
...
         return None
+
+
+def save_yaml_or_json_to_file(
+    bundle_dir: str,
+    filename: str,
+    file_type: str,
+    data: Any,
+) -> None:
+    with open(
```
Comment (epmog) [inline src/deadline/client/job_bundle/loader.py]: Missing a docstring (and it's public!), we should include something here

File: src/deadline/client/job_bundle/loader.py [hunk 2]
```diff
@@ -116,3 +118,20 @@ def read_yaml_or_json_object(
         return parse_yaml_or_json_content(file_contents, file_type, bundle_dir, filename)
     else:
         return None
+
+
+def save_yaml_or_json_to_file(
```
Comment (epmog) [inline src/deadline/client/job_bundle/loader.py]: Should this be moved to a different module? It's currently under `loader` where everything reads in content. feels like we may need to move it to somewhere like `deadline_yaml_dump` for writing out content. Should its naming be similar, like `yaml_or_json_dump`? Not certain myself. Thoughts?

**Review by epmog — CHANGES_REQUESTED** [review]
Code looks good, just need to ensure it's in the right home! Thanks Stephen

File: src/deadline/client/job_bundle/loader.py (hunk 2 above)
Comment (crowecawcaw) [reply src/deadline/client/job_bundle/loader.py]: Moved!

**Review by lucaseck — DISMISSED** [review]

**Review by epmog — APPROVED** [review]

**Review by lucaseck — APPROVED** [review]

**Pull request merged**